### PR TITLE
refactor: replace `probe` with `driver`

### PR DESCRIPTION
**Pull request by leogr**
**What type of PR is this?**

/kind cleanup

/kind design

/kind documentation

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

/area driver-kmod

/area driver-ebpf

/area libscap

/area libsinsp

> /area tests

> /area proposals

**What this PR does / why we need it**:

This PR follows up #39 and corrects the naming convention according to our official [glossary](https://github.com/falcosecurity/falco/tree/master/brand#glossary).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update(driver)!: `PROBE_NAME` has been renamed to `DRIVER_NAME`
update(driver)!: `PROBE_VERSION` has been renamed to `DRIVER_VERSION`
update(driver)!: `PROBE_DEVICE_NAME` has been renamed to `DRIVER_DEVICE_NAME`
```

### CODE REVIEW

File: proposals/20210818-driver-semver.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ not directly related to libscap itself. E.g. two different consumer releases can
 the same libscap commit and still be unable to share the probes.
 
 2. For the kernel module, there is no version check at all. The driver exposes an ioctl
-to get the probe version (`PPM_IOCTL_GET_PROBE_VERSION`) but it is not used anywhere.
+to get the probe version (`PPM_IOCTL_GET_DRIVER_VERSION`) but it is not used anywhere.
```
Comment (FedeDP) [inline proposals/20210818-driver-semver.md:21]:
```suggestion
to get the module version (`PPM_IOCTL_GET_DRIVER_VERSION`) but it is not used anywhere.
```

Comment (leogr) [reply proposals/20210818-driver-semver.md]: Good catch. However, I prefer to fix that in another PR. I have also reverted the other change in the proposal files.

**Review by FedeDP — APPROVED** [review]
/approve

**Review by jasondellaluce — COMMENTED** [review]
LGTM!

**Review by FedeDP — APPROVED** [review]
/approve

**Review by mstemm — APPROVED** [review]

**Pull request merged**